analyzers/src/SonarAnalyzer.CSharp/Rules/AssignmentInsideSubExpression.cs

```diff
@@ -135,30 +125,30 @@ private static bool TryGetCoalesceExpression(ExpressionSyntax expression, out Bi
             return coalesceExpression != null && coalesceExpression.IsKind(SyntaxKind.CoalesceExpression);
         }
 
-        private static bool IsDirectlyInStatementCondition(ExpressionSyntax expression, ExpressionSyntax topParenthesizedExpression)
-        {
-            return IsDirectlyInStatementCondition<IfStatementSyntax>(topParenthesizedExpression, expression, s => s.Condition) ||
-                IsDirectlyInStatementCondition<ForStatementSyntax>(topParenthesizedExpression, expression, s => s.Condition) ||
-                IsDirectlyInStatementCondition<WhileStatementSyntax>(topParenthesizedExpression, expression, s => s.Condition) ||
-                IsDirectlyInStatementCondition<DoStatementSyntax>(topParenthesizedExpression, expression, s => s.Condition);
-        }
+        private static bool IsDirectlyInStatementCondition(ExpressionSyntax expression, ExpressionSyntax topParenthesizedExpression) =>
+            IsDirectlyInStatementCondition<IfStatementSyntax>(topParenthesizedExpression, expression, s => s.Condition)
+            || IsDirectlyInStatementCondition<ForStatementSyntax>(topParenthesizedExpression, expression, s => s.Condition)
+            || IsDirectlyInStatementCondition<WhileStatementSyntax>(topParenthesizedExpression, expression, s => s.Condition)
+            || IsDirectlyInStatementCondition<DoStatementSyntax>(topParenthesizedExpression, expression, s => s.Condition);
```

Comment:
sorry, I saw now the coverage

could you add a test for `IsDirectlyInStatementCondition<DoStatementSyntax>`?